t/plugin/proxy-rewrite.t

```diff
@@ -403,3 +403,223 @@ host: localhost
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 17: add headers
+--- request
+GET /hello HTTP/1.1
+--- response_body
+uri: /uri/plugin_proxy_rewrite
+host: localhost
+x-api-engine: apisix
+x-real-ip: 127.0.0.1
+--- no_error_log
+[error]
+
+
+
+=== TEST 18: set route(remove headers)
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "proxy-rewrite": {
+                                "uri": "/uri/plugin_proxy_rewrite",
+                                "headers": {
+                                    "X-Api-Test": "null"
+                                }
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 19: remove headers
+--- request
+GET /hello HTTP/1.1
+--- more_headers
+X-Api-Test:hello
+--- response_body
+uri: /uri/plugin_proxy_rewrite
+host: localhost
+x-real-ip: 127.0.0.1
+--- no_error_log
+[error]
+
+
+=== TEST 20: set route(rewrite empty headers)
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "proxy-rewrite": {
+                                "uri": "/uri/plugin_proxy_rewrite",
+                                "headers": {
+                                    "X-Api-Test": "hello"
+                                }
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 19: rewrite empty headers
```

Comment:
should be `TEST 21`.
Please use https://github.com/openresty/openresty-devel-utils/blob/master/reindex to format test cases.